src/Runner.Worker/ExecutionContext.cs

```diff
@@ -581,53 +615,59 @@ public void AddIssue(Issue issue, string logMessage = null)
                 issue.Data["stepNumber"] = _record.Order.ToString();
             }
 
-            if (issue.Type == IssueType.Error)
+            string wellKnownTag = null;
+            Int32 previousCountForIssueType = 0;
+            switch (issue.Type)
             {
-                if (!string.IsNullOrEmpty(logMessage))
-                {
-                    long logLineNumber = Write(WellKnownTags.Error, logMessage);
-                    issue.Data["logFileLineNumber"] = logLineNumber.ToString();
-                }
+                case IssueType.Error:
+                    wellKnownTag = WellKnownTags.Error;
+                    previousCountForIssueType = _record.ErrorCount++;
+                    break;
+                case IssueType.Warning:
+                    wellKnownTag = WellKnownTags.Warning;
+                    previousCountForIssueType = _record.WarningCount++;
+                    break;
+                case IssueType.Notice:
+                    wellKnownTag = WellKnownTags.Notice;
+                    previousCountForIssueType = _record.NoticeCount++;
+                    break;
+            }
 
-                if (_record.ErrorCount < _maxIssueCount)
+            if (!string.IsNullOrEmpty(wellKnownTag))
+            {
+                if (previousCountForIssueType < _maxCountPerIssueType)
                 {
                     _record.Issues.Add(issue);
```

Comment:
do we need to not add to the `_record.Issues` when it's embedded?

Reply:
Good question.

`PublishStepTelemetry` (line 1038 in the new version) retains its long-standing guarantee that only issues on top-level (non-embedded) ExecutionContexts are exfiltrated.

That leaves `JobServerQueue.MergeTimelineRecords` as the only other place where `TimelineRecord.Issues` is accessed in a read operation.

I can't say for sure what the ramifications are of allowing issues from embedded `ExecutionContexts` to remain in "merged" `TimelineRecords`, but historically (it seems), the system has tolerated it?

Reply:
I don't think embedded timeline records should be sent up to the server, so it likely doesn't matter. Only root step level timeline records are sent up. 

That being said, whether or not we should add them to the timeline records for embedded is really up to the implementation, it shouldn't matter either way because it just serves as a vehicle to send them up to the server. For confusion sake, it may make sense to not add them unless there is a specific reason we are adding them.

Reply:
Fixed.  Note that the latest commit also avoids updating the record's Error/Warning/NoticeCount when the ExecutionContext is embedded.

Reply:
### Screenshot of verification run after this fix:

![image](https://user-images.githubusercontent.com/81404201/236801078-f0ec6df3-29e2-4ea2-8dc7-99e99aa1c662.png)

source:  https://github.com/bbq-beets/jww3-2022a/actions/runs/4914104452